### PR TITLE
Updates to debugFillProperties to test all properties in slider.dart and slider_test.dart

### PR DESCRIPTION
**Pull request by sladecapasso**
## Description

* This PR is to add DiagnosticsProperties tests to Material Slider, to address issue #25671

## Related Issues

* No related issues

## Tests

I added the following tests:

* DiagnosticsProperties

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by sladecapasso** [comment]
@clocksmith I think new this PR should fix the errors, but it looks like the build is currently broken. Wasn't sure if this is an outside issue I need to wait for resolution on or caused by the pull request.

File: packages/flutter/test/material/slider_test.dart [hunk 1]
```diff
@@ -1547,4 +1547,36 @@ void main() {
     final RenderBox renderObject = tester.renderObject<RenderBox>(find.byType(Slider));
     expect(renderObject.size.height, 200);
   });
+
+  //DiagnosticsProperties Tests
```
Comment (clocksmith) [inline packages/flutter/test/material/slider_test.dart]: You can probably remove this comment

File: packages/flutter/test/material/slider_test.dart [hunk 2]
```diff
@@ -1547,4 +1547,36 @@ void main() {
...
+      value: 10.0,
+      min: 0.0,
+      max: 100.0,
+      activeColor: Colors.blue,
+      inactiveColor: Colors.blue,
+      label: 'Set a value',
+      onChanged: null,
+    ).debugFillProperties(builder);
+
+    print(Slider);
```
Comment (clocksmith) [inline packages/flutter/test/material/slider_test.dart]: remove print

File: packages/flutter/test/material/slider_test.dart [hunk 3]
```diff
@@ -1547,4 +1547,36 @@ void main() {
...
+        .map((DiagnosticsNode node) => node.toString()).toList();
+
+    expect(description, <String>[
+      'value: 10.0',
+      'min: 1.0',
+      'max: 100.0',
+      'activeColor: Colors.blue',
+      'inactiveColor: Colors.blue',
+      'label: "Set a value"',
+      'OnChanged: null',
```
Comment (clocksmith) [inline packages/flutter/test/material/slider_test.dart]: should this be onChanged?

**Review by clocksmith — CHANGES_REQUESTED** [review]
looking good

**Comment by clocksmith** [comment]
This will also need to update the current debugFillProperties with the new properties

File: packages/flutter/test/material/slider_test.dart [hunk 4]
```diff
@@ -1547,4 +1547,41 @@ void main() {
...
+      value: 50.0,
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
+        .map((DiagnosticsNode node) => node.toString()).toList();
+
+    expect(description, <String>[
+      'value: 50.0',
+      'onChanged: disabled',
```
Comment (clocksmith) [inline packages/flutter/test/material/slider_test.dart]: This can just say "disabled"

**Comment by sladecapasso** [comment]
Hi @HansMuller , wanted to provide an update here since Anthony is OOO. He let me know that you wanted Slider methods that are not required and set to null should not show anything in the debugFillProperties output, so I committed those changes earlier and am just waiting for the checks to finish running (and flutter-build to not be broken).

Still pretty new at Flutter (and Github), so let me know if there are any other changes you think are necessary.

File: packages/flutter/lib/src/material/slider.dart [hunk 5]
```diff
@@ -381,8 +381,16 @@ class Slider extends StatefulWidget {
   void debugFillProperties(DiagnosticPropertiesBuilder properties) {
     super.debugFillProperties(properties);
     properties.add(DoubleProperty('value', value));
+    properties.add(ObjectFlagProperty<ValueChanged<double>>('onChanged', onChanged, ifNull: 'disabled'));
+    properties.add(ObjectFlagProperty<ValueChanged<double>>.has('onChangeStart', onChangeStart));
+    properties.add(ObjectFlagProperty<ValueChanged<double>>.has('onChangeEnd', onChangeEnd));
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/slider.dart]: Nice use of ObjectFlagProperty.has. I didn't know we has that :-)

File: packages/flutter/test/material/slider_test.dart [hunk 6]
```diff
@@ -1547,4 +1547,38 @@ void main() {
...
+      min: 0.0,
+      onChanged: null,
+      onChangeEnd: null,
+      onChangeStart: null,
+      semanticFormatterCallback: null,
+      value: 50.0,
+    ).debugFillProperties(builder);
+
+    final List<String> description = builder.properties
+        .where((DiagnosticsNode node) => !node.isFiltered(DiagnosticLevel.info))
```
Comment (HansMuller) [inline packages/flutter/test/material/slider_test.dart]: two space indent

**Review by HansMuller — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/slider.dart (hunk 5 above)
Comment (sladecapasso) [reply packages/flutter/lib/src/material/slider.dart]: Credit goes to @johnsonmh on that one!

**Pull request merged**